discretisedfield/tools/tools.py

```diff
@@ -117,12 +117,9 @@ def topological_charge_density(field, /, method="continuous"):
 
     of = field.orientation  # unit field - orientation field
 
-    # Spatial axes associated with the first and second vector
-    # dimensions.
-    axis1 = field.vdim_mapping[field.vdims[0]]
-    axis2 = field.vdim_mapping[field.vdims[1]]
-    axis1_idx = field.mesh.region._dim2index(axis1)
-    axis2_idx = field.mesh.region._dim2index(axis2)
+    axis1 = field.mesh.region.dims[0]
+    axis2 = field.mesh.region.dims[1]
+    axis1_idx, axis2_idx = 0, 1
```

Comment:
It looks like `axis1_idx` and `axis2_idx` will remain 0 and 1, respectively, regardless of the field used for calculation of the topological charge density. Maybe we can move this over line 120 and use:
```Python
axis1 = field.mesh.region.dims[axis1_idx]
axis2 = field.mesh.region.dims[axis2_idx]
``` 
Do correct me if I understand this wrong.